app/src/org/commcare/android/resource/installers/XFormAndroidInstaller.java

```diff
@@ -71,10 +71,33 @@ public boolean initialize(AndroidCommCarePlatform platform, boolean isUpgrade) t
             IOException, InvalidReferenceException, InvalidStructureException,
             XmlPullParserException, UnfullfilledRequirementsException {
         super.initialize(platform, isUpgrade);
-        platform.registerXmlns(namespace, formDefId);
+        if (isLatestFormRecord(platform)) {
+            platform.registerXmlns(namespace, formDefId);
+        }
         return true;
     }
 
+    // Returns whether the associated formdefRecord is the one with highest form def resource version for our namespace
+    private boolean isLatestFormRecord(AndroidCommCarePlatform platform) {

```

Comment:
From a consistency / safety perspective, I think it's dangerous to make this a per-resource question like this, since it's hard in a closed-form way to guarantee that there *is* at least one form record per XMLNS.

From you and I's breakdown I think we determined that there's only ever two (or a fixed number) of forms that have the same XMLNS possible anyway, so I'd structure this method differently to return *a* valid form ID (probably the one below) and then check it against this one. 

It's a small, stylistic point, but I think right now it's possible for someone to make an easy mistake that results in there being no forms that think they are the latest (due to nested rules).

Reply:
@ctsims Made a change for this though not totally sure if that's what you meant. I am merging it anyway after testing, though feel free to review when you get back online.